apps/gdal_rasterize_lib.cpp

```diff
@@ -50,17 +50,338 @@
+    // Mutex for burn/3d/a
+    {
+        // Required if options for binary
+        auto &group = argParser->add_mutually_exclusive_group(
+            psOptionsForBinary != nullptr);
+
+        group.add_argument("-burn")
+            .metavar("<value>")
+            .scan<'g', double>()
+            .append()
+            .nargs(argparse::nargs_pattern::at_least_one)
+            .action(
+                [psOptions](const std::string &s)
+                {
+                    char **papszTokens = CSLTokenizeString2(s.c_str(), " ", 0);
+                    for (int i = 0; papszTokens[i] != nullptr; i++)
+                    {
+                        psOptions->adfBurnValues.push_back(
+                            CPLAtof(papszTokens[i]));
+                    }
+                })
+            .help(_("A fixed value to burn into the raster band(s)."));
+
+        group.add_argument("-a")
+            .metavar("<attribute_name>")
+            .store_into(psOptions->osBurnAttribute)
+            .help(_("Name of the field in the input layer to get the burn "
+                    "values from."));
+
+        group.add_argument("-3d")
+            .flag()
+            .store_into(psOptions->b3D)
+            .action(
+                [psOptions](const std::string &)
+                {
+                    psOptions->aosRasterizeOptions = CSLSetNameValue(
+                        psOptions->aosRasterizeOptions, "BURN_VALUE_FROM", "Z");
+                })
+            .help(_("Indicates that a burn value should be extracted from the "
+                    "\"Z\" values of the feature."));
+    }
+
+    argParser->add_argument("-add")
+        .flag()
+        .action(
+            [psOptions](const std::string &)
+            {
+                psOptions->aosRasterizeOptions = CSLSetNameValue(
+                    psOptions->aosRasterizeOptions, "MERGE_ALG", "ADD");
+            })
+        .help(_("Instead of burning a new value, this adds the new value to "
+                "the existing raster."));
+
+    // Undocumented
+    argParser->add_argument("-chunkysize")
+        .flag()
+        .hidden()
+        .action(
+            [psOptions](const std::string &s)
+            {
+                psOptions->aosRasterizeOptions = CSLSetNameValue(
+                    psOptions->aosRasterizeOptions, "CHUNKYSIZE", s.c_str());
+            });
+
+    // Mutex with -sql
+    {
+        auto &group = argParser->add_mutually_exclusive_group(false);
+
+        group.add_argument("-l")
+            .metavar("<layer_name>")
+            .append()
+            .store_into(psOptions->aosLayers)
+            .help(_("Name of the layer(s) to process."));
+
+        group.add_argument("-sql")
+            .metavar("<sql_statement>")
+            .store_into(psOptions->osSQL)
+            .action(
+                [psOptions](const std::string &sql)
+                {
+                    GByte *pabyRet = nullptr;
+                    if (sql.at(0) == '@' &&
```

Comment:
I guess someone could trick to pass an empty string, so:
```suggestion
                    if (!sql.empty() && sql.at(0) == '@' &&
```